sdk/tests/conformance2/extensions/webgl-stencil-texturing.html

```diff
@@ -0,0 +1,270 @@
+                                                   wtu.simpleTextureFragmentShaderESSL300]);
+
+    const readStencilShader = `#version 300 es
+        precision highp float;
+        uniform highp usampler2D tex;
+        in vec2 texCoord;
+        out vec4 out_color;
+        void main() {
+            out_color = vec4(texture(tex, texCoord)) / 255.0;
+        }`;
+    const readStencilProgram = wtu.setupProgram(gl, [wtu.simpleTextureVertexShaderESSL300,
+                                                     readStencilShader]);
+
+    for (const format of formats) {
+        debug("");
+        debug(`Testing depth stencil texture modes with ${format.name}`);
+        const fbo = gl.createFramebuffer();
+        gl.bindFramebuffer(gl.FRAMEBUFFER, fbo);
+
+        const rbo = gl.createRenderbuffer();
+        gl.bindRenderbuffer(gl.RENDERBUFFER, rbo);
+        gl.renderbufferStorage(gl.RENDERBUFFER, gl.RGBA8, 1, 1);
+        gl.framebufferRenderbuffer(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.RENDERBUFFER, rbo);
+
+        const tex = gl.createTexture();
+        gl.bindTexture(gl.TEXTURE_2D, tex);
+        gl.texImage2D(gl.TEXTURE_2D, 0, format.internalFormat, 1, 1, 0, format.format, format.type, null);
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR, "texture created");
+
+        gl.framebufferTexture2D(gl.FRAMEBUFFER, gl.DEPTH_ATTACHMENT, gl.TEXTURE_2D, tex, 0);
+        if (format.format == gl.DEPTH_STENCIL) {
+            gl.framebufferTexture2D(gl.FRAMEBUFFER, gl.STENCIL_ATTACHMENT, gl.TEXTURE_2D, tex, 0);
+        }
+        wtu.framebufferStatusShouldBe(gl, gl.FRAMEBUFFER, gl.FRAMEBUFFER_COMPLETE);
+
+        gl.clear(gl.DEPTH_BUFFER_BIT | gl.STENCIL_BUFFER_BIT);
+        gl.useProgram(drawProgram);
+        wtu.drawUnitQuad(gl);
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR, "no errors after drawing to the depth or depth stencil texture");
+
+        // Detach the depth or depth stencil texture to avoid feedback loop
+        gl.framebufferRenderbuffer(gl.FRAMEBUFFER, gl.DEPTH_STENCIL_ATTACHMENT, gl.RENDERBUFFER, null);
+        wtu.framebufferStatusShouldBe(gl, gl.FRAMEBUFFER, gl.FRAMEBUFFER_COMPLETE);
+
+        const magFilters = ['NEAREST', 'LINEAR'];
+
+        const minFilters = [
+            'NEAREST',
+            'LINEAR',
+            'NEAREST_MIPMAP_NEAREST',
+            'LINEAR_MIPMAP_NEAREST',
+            'NEAREST_MIPMAP_LINEAR',
+            'LINEAR_MIPMAP_LINEAR'
+        ];
+
+        const modes = [
+            [gl.DEPTH_COMPONENT, 'DEPTH_COMPONENT'],
+            [ext.STENCIL_INDEX_WEBGL, 'STENCIL_INDEX_WEBGL']
+        ];
+
+        const programs = [
+            [readDepthProgram, 'depth'],
+            [readStencilProgram, 'stencil']
+        ];
+
+        function validFilters(magFilter, minFilter) {
+            return magFilter == gl.NEAREST &&
+                   (minFilter == gl.NEAREST || minFilter == gl.NEAREST_MIPMAP_NEAREST);
```

Comment:
Glad that you're testing that e.g. linear filtering isn't supported, and that the texture is incomplete if using other filters than these. Linear filtering was supported for depth textures as undefined behavior in core OpenGL ES, and some developers started relying on it and filed bugs about its absence on certain GPUs.